Flaky ES test - TestSpanReader_multiRead_followUp_query

defined in plugin/storage/es/spanstore/reader_test.go:208

```
=== RUN   TestSpanReader_multiRead_followUp_query
--- FAIL: TestSpanReader_multiRead_followUp_query (0.00s)
panic: 
mock: Unexpected Method Call
-----------------------------
Index(string,string)
		0: "jaeger-span-2019-08-06"
		1: "jaeger-span-2019-08-05"
The closest call I have is: 
Index(mock.AnythingOfTypeArgument)
		0: "string"
Provided 1 arguments, mocked for 2 arguments
Diff: 1: FAIL:  (string=jaeger-span-2019-08-05) != (Missing) [recovered]
	panic: 
mock: Unexpected Method Call
-----------------------------
Index(string,string)
		0: "jaeger-span-2019-08-06"
		1: "jaeger-span-2019-08-05"
The closest call I have is: 
Index(mock.AnythingOfTypeArgument)
		0: "string"
Provided 1 arguments, mocked for 2 arguments
Diff: 1: FAIL:  (string=jaeger-span-2019-08-05) != (Missing)
goroutine 22 [running]:
testing.tRunner.func1(0xc00010a700)
	/home/travis/.gimme/versions/go1.12.1.linux.amd64/src/testing/testing.go:830 +0x69d
panic(0xca69c0, 0xc000135c00)
	/home/travis/.gimme/versions/go1.12.1.linux.amd64/src/runtime/panic.go:522 +0x1b5
github.com/jaegertracing/jaeger/vendor/github.com/stretchr/testify/mock.(*Mock).fail(0xc0000d5180, 0xd9d701, 0x6f, 0xc000139b40, 0x4, 0x4)
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/stretchr/testify/mock/mock.go:236 +0x1ad
github.com/jaegertracing/jaeger/vendor/github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc0000d5180, 0x1394873, 0x5, 0xc00016c2c0, 0x2, 0x2, 0xc000139b00, 0x4, 0x4)
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/stretchr/testify/mock/mock.go:358 +0xab5
github.com/jaegertracing/jaeger/vendor/github.com/stretchr/testify/mock.(*Mock).Called(0xc0000d5180, 0xc00016c2c0, 0x2, 0x2, 0xc00016c2a0, 0x2, 0x2)
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/stretchr/testify/mock/mock.go:334 +0x1e7
github.com/jaegertracing/jaeger/pkg/es/mocks.(*MultiSearchService).Index(0xc0000d5180, 0xc00016c180, 0x2, 0x2, 0x2b1869d6f208, 0xc0000d5180)
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/es/mocks/MultiSearchService.go:83 +0x1c6
github.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanReader).multiRead(0xc000132c80, 0xe6f5c0, 0xc0000a2010, 0xc0000d7ac0, 0x2, 0x2, 0xbf4a4c71f26eaa02, 0x8a7f75, 0x1408be0, 0xbf4a4c71f26eaa02, ...)
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/reader.go:340 +0xc94
github.com/jaegertracing/jaeger/plugin/storage/es/spanstore.TestSpanReader_multiRead_followUp_query.func1(0xc0000b3f40)
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/reader_test.go:269 +0x21f7
github.com/jaegertracing/jaeger/plugin/storage/es/spanstore.withSpanReader(0xc000060f50)
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/reader_test.go:105 +0x382
github.com/jaegertracing/jaeger/plugin/storage/es/spanstore.TestSpanReader_multiRead_followUp_query(0xc00010a700)
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/reader_test.go:209 +0x57
testing.tRunner(0xc00010a700, 0xda1cb8)
	/home/travis/.gimme/versions/go1.12.1.linux.amd64/src/testing/testing.go:865 +0x164
created by testing.(*T).Run
	/home/travis/.gimme/versions/go1.12.1.linux.amd64/src/testing/testing.go:916 +0x65b
FAIL	github.com/jaegertracing/jaeger/plugin/storage/es/spanstore	0.022s
```

--- a/plugin/storage/es/mappings/gen_assets.go
+++ b/plugin/storage/es/mappings/gen_assets.go
@@ -213,7 +213,7 @@ var _escData = map[string]*_escFile{
 		name:    ".nocover",
 		local:   "plugin/storage/es/mappings/.nocover",
 		size:    43,
-		modtime: 1563813000,
+		modtime: 1563791427,
 		compressed: `
 H4sIAAAAAAAC/youSSzJzFYoSEzOTkxPVcjILy4pVkgsLcnXTU/NSy1KLElNUUjLzEkt1uMCBAAA//8y
 IKK1KwAAAA==
@@ -224,7 +224,7 @@ IKK1KwAAAA==
 		name:    "jaeger-service-7.json",
 		local:   "plugin/storage/es/mappings/jaeger-service-7.json",
 		size:    878,
-		modtime: 1566212432,
+		modtime: 1566472262,
 		compressed: `
 H4sIAAAAAAAC/8ySwW7aQBCG736K1agnBFZViR72RluqVmppBcopikaDPdibeNeb3YEEIb97ZGTAhHDL
 IReP5H++35/t3SVKgXE5P6MnEQ4uglYwuCcuOIwih43JeDSAYbsYWcS4IoJuuQOZurVdcsB6hbGkkLcN
@@ -239,7 +239,7 @@ FMjWM2h44O1THXIYnqemcHVgpGW9YdBfxl97cdPfBU9SoiXJStAgVKQDOMZN8oroOftQZxzjh9DuXNJr
 		name:    "jaeger-service.json",
 		local:   "plugin/storage/es/mappings/jaeger-service.json",
 		size:    1060,
-		modtime: 1565794261,
+		modtime: 1566229177,
 		compressed: `
 H4sIAAAAAAAC/8yTT2/UMBDF7/kU1ojTamshpHLwrUARSFDQVpwQGs3Gs1mD7RjbKayqfHfk4pKkW+2J
 Q3PIn/F78/xz7NtGCMjsgqXMoASsvhN3HM8SxxvT8tkK1kWSOGfjuwSqOIQA4zX/ln5wW47Y7zDtKeoE
@@ -255,7 +255,7 @@ TmGdQJrhQAmkbHr//7o382e5j83Y/AkAAP//qd2MzCQEAAA=
 		name:    "jaeger-span-7.json",
 		local:   "plugin/storage/es/mappings/jaeger-span-7.json",
 		size:    3420,
-		modtime: 1566212432,
+		modtime: 1566472262,
 		compressed: `
 H4sIAAAAAAAC/+xWXW+UQBR951eQG5+aLTEm9YG3amtsYqtp65Mxk7twYaedL2fuVjcN/93A0hYKbE2k
 xhhflix3zuHcmXsO3EZxDNLk9EM4ZCZvAqQx7F0hleT3g0OzvweLuF4WiFmaMkBao+5wiVnrJXlhCxFW
@@ -274,7 +274,7 @@ s/IPX+q/S7/jOB55dNyfveXPdm4jPpxtT0d9N2fQzT1xE5+s9W8VVdHPAAAA//+SuQbQXA0AAA==
 		name:    "jaeger-span.json",
 		local:   "plugin/storage/es/mappings/jaeger-span.json",
 		size:    3830,
-		modtime: 1565794261,
+		modtime: 1566229177,
 		compressed: `
 H4sIAAAAAAAC/+xW0W/TPhB+z18RnX5PUxf9hDQe8jbYEJPYQNt4Qsi6JpfUm2Mb+zqopv7vKE1Lm9ZJ
 QGoQEvShbWx/391n333xcxTHwFRZhUyQxnDygFSSO/UW9ekJTOp5T8xSlx7Senkcg9Q5fUv0vJqSE6YQ

--- a/plugin/storage/es/spanstore/reader_test.go
+++ b/plugin/storage/es/spanstore/reader_test.go
@@ -143,7 +143,7 @@ func TestSpanReaderIndices(t *testing.T) {
 	client := &mocks.Client{}
 	logger, _ := testutils.NewLogger()
 	metricsFactory := metricstest.NewFactory(0)
-	date := time.Now()
+	date := time.Date(2019, 10, 10, 5, 0 ,0 , 0, time.UTC)
 	dateFormat := date.UTC().Format("2006-01-02")
 	testCases := []struct {
 		index  string
@@ -208,22 +208,22 @@ func TestSpanReader_GetTrace(t *testing.T) {
 
 func TestSpanReader_multiRead_followUp_query(t *testing.T) {
 	withSpanReader(func(r *spanReaderTest) {
-		now := time.Now()
-		spanID1 := dbmodel.Span{SpanID: "0", TraceID: "1", StartTime: model.TimeAsEpochMicroseconds(now)}
+		date := time.Date(2019, 10, 10, 5, 0 ,0 , 0, time.UTC)
+		spanID1 := dbmodel.Span{SpanID: "0", TraceID: "1", StartTime: model.TimeAsEpochMicroseconds(date)}
 		spanBytesID1, err := json.Marshal(spanID1)
 		require.NoError(t, err)
-		spanID2 := dbmodel.Span{SpanID: "0", TraceID: "2", StartTime: model.TimeAsEpochMicroseconds(now)}
+		spanID2 := dbmodel.Span{SpanID: "0", TraceID: "2", StartTime: model.TimeAsEpochMicroseconds(date)}
 		spanBytesID2, err := json.Marshal(spanID2)
 		require.NoError(t, err)
 
 		id1Query := elastic.NewTermQuery("traceID", model.TraceID{High: 0, Low: 1}.String())
 		id1Search := elastic.NewSearchRequest().
 			IgnoreUnavailable(true).
-			Source(r.reader.sourceFn(id1Query, model.TimeAsEpochMicroseconds(now.Add(-time.Hour))))
+			Source(r.reader.sourceFn(id1Query, model.TimeAsEpochMicroseconds(date.Add(-time.Hour))))
 		id2Query := elastic.NewTermQuery("traceID", model.TraceID{High: 0, Low: 2}.String())
 		id2Search := elastic.NewSearchRequest().
 			IgnoreUnavailable(true).
-			Source(r.reader.sourceFn(id2Query, model.TimeAsEpochMicroseconds(now.Add(-time.Hour))))
+			Source(r.reader.sourceFn(id2Query, model.TimeAsEpochMicroseconds(date.Add(-time.Hour))))
 		id1SearchSpanTime := elastic.NewSearchRequest().
 			IgnoreUnavailable(true).
 			Source(r.reader.sourceFn(id1Query, spanID1.StartTime))
@@ -264,7 +264,7 @@ func TestSpanReader_multiRead_followUp_query(t *testing.T) {
 				},
 			}, nil)
 
-		traces, err := r.reader.multiRead(context.Background(), []model.TraceID{{High: 0, Low: 1}, {High: 0, Low: 2}}, now, now)
+		traces, err := r.reader.multiRead(context.Background(), []model.TraceID{{High: 0, Low: 1}, {High: 0, Low: 2}}, date, date)
 		require.NoError(t, err)
 		require.NotNil(t, traces)
 		require.Len(t, traces, 2)

--- a/plugin/storage/es/spanstore/reader_test.go
+++ b/plugin/storage/es/spanstore/reader_test.go
@@ -143,7 +143,7 @@ func TestSpanReaderIndices(t *testing.T) {
 	client := &mocks.Client{}
 	logger, _ := testutils.NewLogger()
 	metricsFactory := metricstest.NewFactory(0)
-	date := time.Date(2019, 10, 10, 5, 0 ,0 , 0, time.UTC)
+	date := time.Date(2019, 10, 10, 5, 0, 0, 0, time.UTC)
 	dateFormat := date.UTC().Format("2006-01-02")
 	testCases := []struct {
 		index  string
@@ -208,7 +208,7 @@ func TestSpanReader_GetTrace(t *testing.T) {
 
 func TestSpanReader_multiRead_followUp_query(t *testing.T) {
 	withSpanReader(func(r *spanReaderTest) {
-		date := time.Date(2019, 10, 10, 5, 0 ,0 , 0, time.UTC)
+		date := time.Date(2019, 10, 10, 5, 0, 0, 0, time.UTC)
 		spanID1 := dbmodel.Span{SpanID: "0", TraceID: "1", StartTime: model.TimeAsEpochMicroseconds(date)}
 		spanBytesID1, err := json.Marshal(spanID1)
 		require.NoError(t, err)

